File: Arrays/Majority Element.java
class Solution {
    public int majorityElement(int[] nums) {
        int count = 0, x = nums[0];
        for (int num : nums) {
            if (num == x)
                count++;
            else {
                count--;
                if (count == 0) {
                    x = num;
                    count++;
                }
            }
        }
        return x;
    }
}

File: Arrays/Majority Element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        x, count = nums[0], 0
        for v in nums:
            if v == x:
                count += 1
            else:
                count -= 1
                if count == 0:
                    x = v
                    count = 1
        return x
